### DISCUSSION TITLE
Does rgb support transfer the asset to taproot sh or wsh?

### DISCUSSION THREAD

**Discussion by Crayon-Shin-chan-bitlightlabs** [RGB-WG/rgb#266] [General]
For example, locking RGB assets into an HTLC script, where ownership of the assets is obtained only by unlocking the HTLC script’s output. Or does RGB have other methods to achieve such a requirement?

**Comment by St333p** [comment]
I'm not sure I understand your question, but it is indeed possible to allocate RGB assets to any bitcoin UTXO, regardless of the script locking it. This includes HTLC scripts, which are necessary for RGB on LN.

**Reply by Crayon-Shin-chan-bitlightlabs** [reply]
> I'm not sure I understand your question, but it is indeed possible to allocate RGB assets to any bitcoin UTXO, regardless of the script locking it. This includes HTLC scripts, which are necessary for RGB on LN.

i think so, but he bp-wallet don't support any wsh or taproot script descriptor, so the wallet can't spent the UTXO

**Reply by St333p** [reply]
As far as I can see, `bp-wallet` takes a generic for the descriptor, so it should be possible to pass it different ones from the two defined in the `StdDescr` enum. In any case, it's always possible to build an RGB wallet using libraries other than `bp-wallet`.

**Reply by dr-orlovsky** [reply]
Exactly. It takes anything what implements `Descriptor` trait. You either may add `Wsh` and `Sh` to the `StdDescr` in a PR to `bp-std` lib (desired way) - or do your own custom descriptor type as you see.

I had plans to add them anyway, just no time yet...
